indexing-hadoop/src/main/java/io/druid/indexer/IndexGeneratorJob.java

```diff
@@ -524,6 +541,35 @@ protected void reduce(
 
         Set<String> allDimensionNames = Sets.newHashSet();
         final ProgressIndicator progressIndicator = makeProgressIndicator(context);
+        int persistBackgroundCount = config.getSchema().getTuningConfig().getPersistBackgroundCount();
+        if (persistBackgroundCount > 0) {
+          final BlockingQueue<Runnable> queue = new SynchronousQueue<>();
+          ExecutorService executorService = new ThreadPoolExecutor(
+              persistBackgroundCount,
+              persistBackgroundCount,
+              0L,
+              TimeUnit.MILLISECONDS,
+              queue,
+              Execs.makeThreadFactory("IndexGeneratorJob_persist_%d"),
+              new RejectedExecutionHandler()
```

Comment:
Execs.newBlockingSingleThreaded(..) only have one background thread to persist incremental Index, so i have not use it.